sphinx_gallery/tests/test_notebook.py

```diff
@@ -66,3 +75,39 @@ def test_convert():
 ![me](foobar)
 """  # noqa
     assert_equal(rst2md(rst), markdown)
+
+
+def test_jupyter_notebook():
+    """Test that written ipython notebook file corresponds to python object"""
+    blocks = sg.split_code_and_text_blocks('tutorials/plot_parse.py')
+    example_nb = jupyter_notebook(blocks)
+
+    with tempfile.NamedTemporaryFile('w') as nb_file:
+        save_notebook(example_nb, nb_file.name)
+        with open(nb_file.name, "r") as fname:
+            assert_equal(json.load(fname), example_nb)
+
+###############################################################################
+# Notebook shell utility
+
+
+class CommandLineTest(TestCase):
+    """Test the Sphinx-Gallery python to Jupyter notebook converter CLI"""
+
+    def test_with_empty_args(self):
```

Comment:
I push a change in your branch because apart from this it is good for merging I think.